lib/drivers/bayer/bayerContourNext.js

```diff
@@ -149,7 +150,11 @@ module.exports = function (config) {
   var parseHeader = function (header, callback){
     var pString = header.split('|');
     var pInfo = pString[4].split('^');
-    var sNum = pInfo[2].match(/^\d+\-\s*(\w+)/);
+
+    // serial number can be in two formats:
+    // 6301-1C2CF8C -> C2CF8C
+    // 7830H5001733 -> 5001733
+    var sNum = pInfo[2].match(/^\d+\-?\w?\s*(\w+)/);
```

Comment:
This won't work for the old format (see comment above about original regex). Suggest `/^\d+[\w-]\s*(\w+)/`.
I'm also not quite sure why the `\s*` is in there, because I don't see any white space. But I'll assume that it must have occurred in an instance in the past for it to be included in the regex.

Reply:
I have not seen the white space either, but I also made the same assumption. So here is what happened when I uploaded the 11 meters (all the different models we support, with some mmol/L versions as well) I have here, using the regex you suggested: 10 meters have a correct serial number, except for the BCN Link with serial number C2CF8C, where 6301-1C2CF8C -> 1C2CF8C. 

According to the spec (which is publicly available, so I'll past the relevant sections here):

> Note that the serial number is actually the model number followed by the serial number field. Additionally, the model number can have a 5th character.

> Meter serial number (7350-SSSSSSS). The first 5 characters represent the meter model
number (the dash " – " can be replaced with a capital letter for derived product codes

So according to the spec it seems we should be storing that extra `1` as part of the serial number, even if that is not what is written on the physical device.